vendor/knative.dev/pkg/tracing/opencensus.go

```diff
@@ -120,15 +126,25 @@ func WithExporter(name string, logger *zap.SugaredLogger) ConfigOption {
 			})
 			if err != nil {
 				logger.Errorw("error reading project-id from metadata", zap.Error(err))
-				return
+				return err
 			}
 			exporter = exp
 		case config.Zipkin:
+			// If name isn't specified, then zipkin.NewEndpoint will return an error saying that it
+			// can't find the host named ''. So, if not specified, default it to this machine's
+			// hostname.
+			if name == "" {
+				n, err := os.Hostname()
+				if err != nil {
+					return fmt.Errorf("unable to get hostname: %v", err)
+				}
+				name = n
+			}
```

Comment:
I think all the changes to ` vendor/knative.dev/pkg/tracing/` should be submit to knative pkg repo. Or else, it will be overwrote when updating the dependencies.

Reply:
Absolutely, https://github.com/knative/pkg/pull/658 is making the change in pkg.